stripe/src/main/java/com/stripe/android/StripeKtx.kt

```diff
@@ -0,0 +1,581 @@
+ *
+ * See [Confirm a SetupIntent](https://stripe.com/docs/api/setup_intents/confirm).
+ * `POST /v1/setup_intents/:id/confirm`
+ *
+ * @param confirmSetupIntentParams a set of params with which to confirm the Setup Intent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [SetupIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmSetupIntentSuspend(
+    confirmSetupIntentParams: ConfirmSetupIntentParams,
+    idempotencyKey: String? = null
+): SetupIntent = runOnIOAndThrowIfNull {
+    stripeRepository.confirmSetupIntent(
+        confirmSetupIntentParams,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId,
+            idempotencyKey = idempotencyKey
+        )
+    )
+}
+
+/**
+ * Suspend function to confirm a [PaymentIntent] object.
+ *
+ * See [Confirm a PaymentIntent](https://stripe.com/docs/api/payment_intents/confirm).
+ * `POST /v1/payment_intents/:id/confirm`
+ *
+ * @param confirmPaymentIntentParams a set of params with which to confirm the PaymentIntent
+ * @param idempotencyKey optional, see [Idempotent Requests](https://stripe.com/docs/api/idempotent_requests)
+ *
+ * @return a [PaymentIntent] object
+ *
+ * @throws AuthenticationException failure to properly authenticate yourself (check your key)
+ * @throws InvalidRequestException your request has invalid parameters
+ * @throws APIConnectionException failure to connect to Stripe's API
+ * @throws APIException any other type of problem (for instance, a temporary issue with Stripe's servers)
+ */
+@Throws(
+    AuthenticationException::class,
+    InvalidRequestException::class,
+    APIConnectionException::class,
+    APIException::class
+)
+suspend fun Stripe.confirmPaymentIntentSuspend(
+    confirmPaymentIntentParams: ConfirmPaymentIntentParams,
+    idempotencyKey: String? = null
+): PaymentIntent = runOnIOAndThrowIfNull {
+    stripeRepository.confirmPaymentIntent(
+        confirmPaymentIntentParams,
+        ApiRequest.Options(
+            apiKey = publishableKey,
+            stripeAccount = stripeAccountId,
+            idempotencyKey = idempotencyKey
+        )
+    )
+}
+
+/**
+ * Consume the empty result from Stripe's internal Json Parser, throw [InvalidRequestException] for public API.
+ */
+internal suspend inline fun <reified T : Any> runOnIOAndThrowIfNull(
+    crossinline block: suspend () -> T?
+): T =
+    withContext(Dispatchers.IO) {
```

Comment:
`workContext`

Reply:
I think we don't need to switch the context here. The rule of thumb is to let the component responsible for doing the work to switch to the correct context. `StripeRepository` already defines all its public methods as suspend functions and `StripeApiRepository` handles the switch on `fetchStripeModel(...)`, `makeApiRequest(...)`, and `makeFileUploadRequest(...)`. In this case we have the `Stripe` extension switching to a dispatcher and `StripeApiRepository` also switching to a dispatcher. We could even push the dispatcher further down into the request executor as that is the component doing the actual network work.

Reply:
➕ to this

Reply:
See https://github.com/stripe/stripe-android/pull/3561